### PR TITLE
fix: filter packages not installed for the current system

### PR DESCRIPTION
**Pull request by ysndr**
Detect missing packages for the current system in the lockfile
and omit them when listing.

Future changes could change the `list` output to include these packages
with a note that they are _not_ installed for the current system.

closes flox/flox#761

### CODE REVIEW

File: cli/tests/environment-list.bats [hunk 1]
```diff
@@ -99,3 +99,26 @@ EOF
...
+    htop = { path = "htop", systems = [] }
+EOF
+
+  )"
+
+  echo "$MANIFEST_CONTENT" | "$FLOX_BIN" edit -f -
+
+  run "$FLOX_BIN" list -n
+  assert_success
+  assert_output "hello"
```
Comment (mkenigs) [inline cli/tests/environment-list.bats:123]: suggestion blocking: refute htop in the output

**Review by mkenigs — CHANGES_REQUESTED** [review]

File: cli/tests/environment-list.bats (hunk 1 above)
Comment (mkenigs) [reply cli/tests/environment-list.bats:123]: Or change the test description to say that list doesn't fail or something

Comment (ysndr) [reply cli/tests/environment-list.bats:123]: doesnt assert_output "assert the entire output" (esp. without `--partial`)?

Comment (mkenigs) [reply cli/tests/environment-list.bats:123]: Oh you're right sorry I misread that

**Review by mkenigs — APPROVED** [review]

**Pull request merged**